File: StraitOfHormuz/interface/Dockerfile.go
FROM golang:1.26-alpine AS builder

WORKDIR /app

COPY go.mod go.sum ./
RUN go mod download

COPY ./interface/interface.go ./interface.go

RUN CGO_ENABLED=0 GOOS=linux go build -o interface_bin interface.go

FROM alpine:latest

WORKDIR /app

COPY --from=builder /app/interface_bin .
COPY ./data/initialization/interface.json ./data/initialization/interface.json

CMD ["./interface_bin"]

File: StraitOfHormuz/interface/Dockerfile.py
FROM python:3.11-alpine

WORKDIR /app

RUN apk add --no-cache build-base sdl2-dev sdl2_image-dev sdl2_mixer-dev sdl2_ttf-dev

COPY ./interface/interface.py ./interface.py

COPY ./data/images ./data/images
COPY ./data/interface ./data/interface

RUN pip install --no-cache-dir pygame

CMD ["python3", "interface.py"]